hubblestack/extmods/grains/hostuuid.py

```diff
@@ -15,14 +15,36 @@ def host_uuid():
     Generate a unique uuid for this host, storing it on disk so it persists
     across restarts
     '''
-    cached_uuid = os.path.join(os.path.dirname(__opts__['configfile']), 'hubble_cached_uuid')
+    cached_uuid_path = os.path.join(os.path.dirname(__opts__['configfile']), 'hubble_cached_uuid')
+    existing_uuid = __opts__.get('host_uuid', None)
     try:
-        if os.path.isfile(cached_uuid):
-            with open(cached_uuid, 'r') as f:
-                return {'host_uuid': f.read()}
-    except Exception as exc:
-        log.exception('Problem retrieving cached host uuid')
-    generated = str(uuid.uuid4())
-    with open(cached_uuid, 'w') as f:
-        f.write(generated)
-    return {'host_uuid': generated}
+        if os.path.isfile(cached_uuid_path):
+            with open(cached_uuid_path, 'r') as f:
+                cached_uuid = f.read()
+                # Check if it's changed out from under us -- problem!
+                if existing_uuid and cached_uuid != existing_uuid:
+                    log.error('host_uuid changed on disk unexpectedly!'
+                              '\nPrevious: {0}\nNew: {1}\nKeeping previous.'
+                              .format(existing_uuid, cached_uuid))
+                    return {'host_uuid': existing_uuid}
+                return {'host_uuid': cached_uuid}
+        elif existing_uuid:
+            log.error('host_uuid was previously generated, but the cached '
+                      'file is no longer present: {0}'.format(cached_uuid_path))
+    except Exception:
+        log.exception('Problem retrieving cached host uuid from file: {0}'
+                      .format(cached_uuid_path))
+
+    # Generate a fresh one if needed
+    if not existing_uuid:
+        existing_uuid = str(uuid.uuid4())
```

Comment:
Can we also add a log message on generating a fresh one? 
Chances are that this should fire rarely. When it does fire, either there was a clean wipe (stop & cache delete - something we'd like to be aware of) or it's a completely new host - also a cause for celebration.

Reply:
Done. I added it further up, with the rest of the messages around the cache file: https://github.com/hubblestack/hubble/pull/260/commits/d9fe303f5ef4d754e08dd34b5718f82f5866ee8b